libs/template/schema.go

```diff
@@ -0,0 +1,145 @@
+package template
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"reflect"
+)
+
+// This is a JSON Schema compliant struct that we use to do validation checks on
+// the provided configuration
```

Comment:
We also have a schema in `bundle/schema`. Could we consolidate into using a single JSON schema type and using it from both places? The redundancy seems not warranted.

Reply:
done